packages/ui/src/components/catalog/Catalog.jsx

```diff
@@ -1,107 +1,205 @@
-import { useState } from "react";
+import { useEffect, useMemo, useState } from "react";
 import leftArrow from "../../assets/left-arrow.svg";
 import rightArrow from "../../assets/right-arrow.svg";
-import searchLogo from "../../assets/searchIcon.svg";
-import { companies } from "../../utils/Constants";
 import styles from "./Catalog.module.css";
 import CatalogItem from "./CatalogItem";
 import ImageUploadModal from "./ImageUploadModal";
 import CustomInput from "../common/input/CustomInput";
 import Button from "../common/button/Button";
+import { useApi } from "../../hooks/useApi";
+import { useToast } from "../../hooks/useToast";
+import LoadingSpinner from "../common/loadingspinner/LoadingSpinner.jsx";
+import { MESSAGES } from "../../utils/Constants.js";
 
 function Catalog() {
+  const toast = useToast();
   const [pageNum, setPageNum] = useState(0);
   const [searchTerm, setSearchTerm] = useState("");
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [totalPages, setTotalPages] = useState(0);
+  const [updateImageId, setUpdateImageId] = useState(null);
 
   const limit = 10;
-  const totalPages = Math.floor(companies.length / 10);
+  const skip = pageNum * limit;
 
-  const skip = pageNum;
-  const skipCount = skip * limit;
+  const { data, loading, makeRequest, errorMsg } = useApi({
+    method: "GET",
+    url: `/catalog/logos?skip=${skip}&limit=${limit}`,
+  });
 
-  const companiesInfo = companies.slice(
-    skipCount,
-    Math.min(skipCount + limit, companies.length)
-  );
-  const filteredCompanies = companiesInfo.filter((company) =>
-    company.companyImage.toLowerCase().includes(searchTerm.toLowerCase())
-  );
+  const {
+    loading: uploadLoading,
+    makeRequest: uploadMakeRequest,
+    errorMsg: uploadErrorMsg,
+  } = useApi({
+    method: updateImageId ? "PUT" : "POST",
+    url: `/catalog/logo`,
+    headers: { "Content-Type": "multipart/form-data" },
+  });
+
+  useEffect(() => {
+    makeRequest();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pageNum]);
```

Comment:
is the interface going to be an infinite scroll pagination? If not then why are we not having support for url search params which will allow the user to have browser navigation controls